### DISCUSSION TITLE
How does this compare to Prismatik & Arduino Nano running Adalight?

### DISCUSSION THREAD

**Discussion by oleost** [awawa-dev/HyperHDR#836] [Q&A] [answered]
Hello.

I have WS2812B LEDs powered by an Arduino Nano with https://github.com/Wifsimster/adalight_ws2812 installed, which is commanded by https://github.com/psieg/Lightpack so that I have Ambilight functionality on my PC screen.

Wondering how this project pared with an ESP32 running https://github.com/awawa-dev/HyperSerialESP32 compares to the previous mentioned combination?

Anyone have any experiences regarding the use of these? 
The only reason I might consider switching is if there is slightly less delay and a higher update frequency with the ESP32 solution.

**Comment by awawa-dev** [comment, marked as answer]
Hi
In our projects, we use a much higher speed, which naturally favors the occurrence of various types of errors during transmission (according to the documentation of various UARTs up to several percent, usually it is less, but with standard Adalight it would result in artifacts on the LED strip), but our protocol extended to include data integrity checking allows detect them and ignore damaged frames. Additionally, it offers the ability to calibrate the white channel for RGBW sk6812. If you decide on ESP32 and HyperSerialESP32, remember that the UART must support a speed of 2Mb or more: when ordering from Aliexpress, for example, you never know what you will get. In turn, rp2040 and HyperSerialPico always provide the right speed and it easier to flash it.

If you want to read about the origins of the HyperSerial projects, you can find background story and more details here:
https://github.com/awawa-dev/HyperSerialEsp8266
